app/services/PayeAtsService.scala

```diff
@@ -46,19 +48,46 @@ class PayeAtsService @Inject()(middleConnector: MiddleConnector, auditService: A
         Left(HttpResponse(INTERNAL_SERVER_ERROR, e.getMessage))
     }
 
-  def getPayeATSMultipleYearData(nino: Nino, yearFrom: Int, yearTo: Int)(
+  def getPayeTaxYearData(nino: Nino, yearFrom: Int, yearTo: Int)(
     implicit hc: HeaderCarrier,
-    request: PayeAuthenticatedRequest[_]): Future[Either[HttpResponse, List[PayeAtsData]]] =
+    request: AuthenticatedRequest[_]): Future[Either[HttpResponse, List[Int]]] =
     middleConnector.connectToPayeATSMultipleYears(nino, yearFrom, yearTo) map { response =>
-      handleConnectorResponse[List[PayeAtsData]](response, nino, yearFrom)
+      handlePayeTaxYearDataResponse[List[PayeAtsData]](response, nino, yearFrom)
     } recover {
       case e: BadRequestException => Left(HttpResponse(BAD_REQUEST, e.getMessage))
-      case e: NotFoundException   => Left(HttpResponse(NOT_FOUND, e.getMessage))
+      case e: NotFoundException   => Right(List.empty)
       case e: Exception =>
         Logger.error(s"Exception in PayeAtsService: $e", e)
         Left(HttpResponse(INTERNAL_SERVER_ERROR, e.getMessage))
     }
 
+  private def handlePayeTaxYearDataResponse[A](response: HttpResponse, nino: Nino, taxYear: Int)(
+    implicit reads: Reads[List[PayeAtsData]],
+    hc: HeaderCarrier,
+    request: AuthenticatedRequest[_]): Either[HttpResponse, List[Int]] =
+    response.status match {
+      case OK =>
+        sendAuditEventForPayeTaxYearData(nino, taxYear)
+        val res = response.json.as[List[PayeAtsData]]
+        Right(res.map(_.taxYear).reverse)
+      case NOT_FOUND => Right(List.empty)
+      case _ => {
+        logger.error(s"Error received, Http status: ${response.status}")
+        Left(response)
+      }
+    }
+
+  private def sendAuditEventForPayeTaxYearData(nino: Nino, taxYear: Int)(
+    implicit hc: HeaderCarrier,
+    request: AuthenticatedRequest[_]): Future[AuditResult] =
+    auditService.sendEvent(
```

Comment:
All new audit events need to be cleared with CIP team, not sure we even need to audit the block above? IT'll be implicitly audited in connectToPayeATSMultipleYears

Reply:
removed